### ISSUE TITLE
S3CMD --configure throws error

### ISSUE THREAD

**Issue by usagar80** [s3tools/s3cmd#1356]
In mac, after installing latest `s3cmd` when I try to run configure command, it throws below error

```
ERROR: Test failed: sequence item 1: expected str instance, bytes found
```

![image](https://github.com/s3tools/s3cmd/assets/4917613/cd0d2192-25d8-4c5b-8f50-c899bfad9d55)

Additionally I run `--debug` and saw the AWS send correct respond with info in XML format.

![image](https://github.com/s3tools/s3cmd/assets/4917613/4eeb8d15-bcbc-4a8a-b20a-491a240c8c4d)

**Comment by aweakley** [comment]
I'm getting something similar. This is from a `brew install` of v2.3.0

```
Invoked as: /opt/homebrew/bin/s3cmd ls
Problem: <class 'TypeError: sequence item 1: expected str instance, bytes found
S3cmd:   2.3.0
python:   3.12.0 (main, Oct  2 2023, 12:03:24) [Clang 15.0.0 (clang-1500.0.40.1)]
environment LANG=en_AU.UTF-8

Traceback (most recent call last):
  File "/opt/homebrew/bin/s3cmd", line 3286, in <module>
    rc = main()
         ^^^^^^
  File "/opt/homebrew/bin/s3cmd", line 3183, in main
    rc = cmd_func(args)
         ^^^^^^^^^^^^^^
  File "/opt/homebrew/bin/s3cmd", line 171, in cmd_ls
    subcmd_all_buckets_list(s3)
  File "/opt/homebrew/bin/s3cmd", line 176, in subcmd_all_buckets_list
    response = s3.list_all_buckets()
               ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.12/site-packages/S3/S3.py", line 327, in list_all_buckets
    response["list"] = getListFromXml(response["data"], "Bucket")
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.12/site-packages/S3/BaseUtils.py", line 277, in getListFromXml
    tree = getTreeFromXml(xml)
           ^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.12/site-packages/S3/BaseUtils.py", line 263, in getTreeFromXml
    xml, xmlns = stripNameSpace(encode_to_s3(xml))
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.12/site-packages/S3/BaseUtils.py", line 255, in stripNameSpace
    xml = RE_XML_NAMESPACE.sub("\\1\\2", xml, 1)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: sequence item 1: expected str instance, bytes found
```

**Comment by watou** [comment]
File "/opt/homebrew/lib/python3.12/site-packages/S3/BaseUtils.py", line 255, in stripNameSpace
```
    xml = RE_XML_NAMESPACE.sub("\\1\\2", xml, 1)
```
change to 
```
    xml = RE_XML_NAMESPACE.sub(b"\\1\\2", xml, 1)
```
until a new release is made to work with Python 3.12.  I think!  Worked for me.

**Comment by i-break-codes** [comment]
> File "/opt/homebrew/lib/python3.12/site-packages/S3/BaseUtils.py", line 255, in stripNameSpace
> 
> ```
>     xml = RE_XML_NAMESPACE.sub("\\1\\2", xml, 1)
> ```
> 
> change to
> 
> ```
>     xml = RE_XML_NAMESPACE.sub(b"\\1\\2", xml, 1)
> ```
> 
> until a new release is made to work with Python 3.12. I think! Worked for me.

Ok, this worked for me, thanks for the help bud, appreciated 👍

**Comment by rampageservices** [comment]
This PR fixes this issue on Python 3.12 with s3cmd installed via brew.
https://github.com/s3tools/s3cmd/pull/1355

**Comment by fviard** [comment]
This issue was already fixed in MASTER and there is now also a new release available with this correction: v2.4.0.
